syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";

message Kubectl {
  message Command {
    google.protobuf.StringValue command = 1;
  }

  message WriteCommand {
    google.protobuf.StringValue command = 1;
  }

  enum TaskType {
    UNKNOWN = 0;
    COMMAND = 1;
    K8S_WRITE_COMMAND = 2;
  }

  TaskType type = 1;
  oneof task {
    Command command = 2;
    WriteCommand k8s_write_command = 3;
  }
}
